.github/workflows/clean.yml

```diff
@@ -0,0 +1,39 @@
+name: clean
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  clean:
+    name: clean
+    runs-on: ubuntu-latest
+    steps:
+      - name: Collect Metrics
+        id: collect-gha-metrics
+        uses: smartcontractkit/push-gha-metrics-action@v1
+        with:
+          basic-auth: ${{ secrets.GRAFANA_CLOUD_BASIC_AUTH }}
+          hostname: ${{ secrets.GRAFANA_CLOUD_HOST }}
+          this-job-name: clean
+        continue-on-error: true
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        with:
+          fetch-depth: 0
+      - name: Setup Go
+        uses: ./.github/actions/setup-go
+        with:
+          only-modules: "true"
+      - run: make mockery
+      - run: make abigen
+      - run: make gomodtidy
+      - name: Ensure clean after tidy
+        run: git diff --minimal --exit-code
+      - run: npm install -g pnpm
+      - run: make pnpmdep
```

Comment:
You can use https://github.com/smartcontractkit/chainlink/blob/20bc44b96ff2bb1e2f7eedb36f040c197d915dd1/.github/actions/setup-nodejs/action.yaml#L6 instead